### PR TITLE
Hypershift upgrade modal

### PR DESCRIPTION
**Pull request by fxiang1**
https://issues.redhat.com/browse/ACM-1745

- Create a new modal to support upgrading hypershift clusters
- The control plane and nodepools can be upgraded separately so logic was implemented to support that

image

image

### CODE REVIEW

File: frontend/public/locales/en/translation.json [hunk 1]
```diff
@@ -83,6 +83,7 @@
     "Add credentials and custom certificate authority (CA) certificates to connect to private helm chart repository.": "Add credentials and custom certificate authority (CA) certificates to connect to private helm chart repository.",
     "Add hosts": "Add hosts",
     "Add new resource": "Add new resource",
+    "Add nodepool": "Add nodepool",
     "Add Nodepool": "Add Nodepool",
```
Comment (chenz4027) [inline frontend/public/locales/en/translation.json:87]: If we no longer use the upper case, maybe we should remove it?

Comment (chenz4027) [reply frontend/public/locales/en/translation.json:87]: One thing that I'm concerned is that "Add Nodepool" has translations in Japanese, Chinese and Korean. By changing that we would miss it from existing translations - which is fine as long as we have new translations. I wonder for the texts that are introduced (not just add nodepools, but other texts), is that a way to generate them in those three languages for the PR? 

Language translation PR: https://github.com/stolostron/console/pull/2168

**Review by chenz4027 — COMMENTED** [review]
Also the PR still has WIP in the title ;)

File: frontend/public/locales/en/translation.json (hunk 1 above)
Comment (KevinFCormier) [reply frontend/public/locales/en/translation.json:87]: @chenz4027 You do not need to worry about the non-en languages. We will have additional rounds of translation before the end of the release. This was just an initial batch of translations to make sure the setup is working.

Comment (chenz4027) [reply frontend/public/locales/en/translation.json:87]: Oh perfect, thanks @KevinFCormier 👍 
/lgtm

Comment (fxiang1) [reply frontend/public/locales/en/translation.json]: That's a good catch @chenz4027. I removed the extra the `Add Nodepool` with the upper case.

**Review by chenz4027 — APPROVED** [review]
/lgtm

Thanks Feng

**Pull request merged**